### PR TITLE
Add support for z.catch coercion

### LINKED ISSUE #397: Compatibility issue with `ZodCatch`
### Describe the bug and the expected behavior

If I add a `.catch()` to a `z.number()` schema, `parseWithZod()` always returns the catch value, even with a valid input that matches the Zod schema.

### Conform version

v1.0.0-rc.1

### Steps to Reproduce the Bug or Issue

Minimal reproduction: https://stackblitz.com/edit/remix-run-remix-qxwt5m?file=app%2Froutes%2F_index.tsx

Expected behaviour:

```ts
const schema = z.object({
  page: z.number().int().min(1).safe().catch(1),
});

expect(
  parseWithZod(new URLSearchParams('page=2'), {
    schema,
  })
).toEqual({
  status: 'success',
  payload: {
    page: '2',
  },
  value: {
    page: 2,
  },
  reply: expect.any(Function),
});
```

Actual behaviour: `value` is always equal to `{ page: 1 }`.

### What browsers are you seeing the problem on?

_No response_

### Screenshots or Videos

_No response_

### Additional context

The use case here is validating `searchParams`, where unlike `formData` you probably just want to fall back to valid values and ignore invalid ones.

If you remove the `.catch()` from the schema, it works as expected—but of course then you can't fall back to valid values.

### PR DESCRIPTION
**Pull request by aaronadamsCA**
Fixes #397

### CODE REVIEW

File: tests/conform-zod.spec.ts [hunk 1]
```diff
@@ -869,6 +869,53 @@ describe('conform-zod', () => {
...
+					createFormData([
+						['a', ''],
+						['b', ''],
+						['c', ''],
+						['d', ''],
+						['e', emptyFile],
+						['f', ''],
+					]),
+					{ schema },
+				),
```
Comment (edmundhung) [inline tests/conform-zod.spec.ts:896]: Can you add one more assertion just to verify what happens if the formdata is valid? e.g. `a` is `helloeworld` and `b` is `456`.

Comment (aaronadamsCA) [reply tests/conform-zod.spec.ts:898]: Done.

File: packages/conform-zod/coercion.ts [hunk 2]
```diff
@@ -219,6 +220,15 @@ export function enableTypeCoercion<Schema extends ZodTypeAny>(
...
 			);
+	} else if (type instanceof ZodCatch) {
+		schema = any()
+			.transform((value) => coerceFile(coerceString(value)))
+			.pipe(
+				new ZodCatch({
+					...type._def,
+					innerType: enableTypeCoercion(type.removeCatch(), cache),
+				}),
+			);
```
[the remark this answers is not in the document]
Comment (aaronadamsCA) [reply packages/conform-zod/coercion.ts]: It does! Thanks for that, as it was I just copied the default block to start with.

**Review by edmundhung — APPROVED** [review]
Thanks for working on this! I will probably release it with v1 directly.

**Pull request merged**